File: 219-ContainsDuplicateIi/219-ContainsDuplicateIi.cpp
// Last updated: 6/10/2026, 11:49:42 AM
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        map<int,int> mp;
        int n = nums.size();
        for(int i = 0 ; i < min(k , n); i++){
            mp[nums[i]]++;
            if(mp[nums[i]] == 2)  return 1;
        }
        for(int i = k; i < nums.size(); i++){
            mp[nums[i]]++;
            if(mp[nums[i]] == 2)  return 1;
            mp[nums[i-k]]--;
        }
        return 0;
    }
};

File: 219-ContainsDuplicateIi/219-ContainsDuplicateIi.java
// Last updated: 6/10/2026, 11:49:47 AM
import java.util.Map;
import java.util.HashMap;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer , Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if(map.containsKey(nums[i]))
                {
                    int x = map.get(nums[i]);
                    System.out.println(x + "this is  x \t and i" + i);
                    if((int)Math.abs(x-i) <= k)
                        return true;
                    else
                        map.replace(nums[i], i);
                }
            else
                map.put(nums[i],i);                
        }
        return false;
    }
}